internal/core/id.go

```diff
@@ -2,11 +2,61 @@ package core
 
 import (
 	"fmt"
+
+	"github.com/google/uuid"
 )
 
-// ComponentID ... Represents a deterministic ID that's assigned
-// to all ETL components
-type ComponentID [4]byte
+type UUID struct {
+	uuid.UUID
+}
+
+func newUUID() UUID {
+	return UUID{
+		uuid.New(),
+	}
+}
+
+func nilUUID() UUID {
+	return UUID{[16]byte{0}}
+}
+
+// String ... Overrides UUID string method for easier
+// debugging and ensuring conformance with pessimism specific abstractions
+// https://pkg.go.dev/github.com/google/UUID#UUID.String
+func (id UUID) String() string {
+	uid := id.UUID
+	// Only render first 8 bytes instead of entire sequence
```

Comment:
So, if I understand correctly, we use the first 8 bytes to _examine_ the UUID but still operate with the full underlying 36 bytes in practice? This feels sketchy, but I can't think of a specific reason to oppose. Have you thought about this same concern about a discrepancy between what we use and what the string representation returns?

Reply:
This is just for debugging purposes which is why the string representation is reduced to only 8 byte. Realistically, the likelihood of a conflict with a 2^64 value space is very unlikely.  However, I do agree that `String` can give a negative connotation and this method may be better suited something as something like `ShortString`.

Reply:
Yeah, I'm a strong advocate for `ShortString` if there's no expectation of the `String` method itself being used. Would much prefer that to prevent any misconception that the `String` representation perfectly reflects the actual underlying value. @epociask